tools/display/timestamp_diffs.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+from typing import List
+from matplotlib import pyplot as plt
+import pandas as pd
+import os
+import argparse
+from scipy.signal import find_peaks
+
+
+def plot_statistics_summary(diffs: np.ndarray):
+    """
+    Creates a histogram to visualize the statistical distribution of timestamp differences.
+
+    Parameters:
+    -----------
+        diffs (np.ndarray): The array of timestamp differences.
+    """
+    # 1. Calculate Statistics
+    mean_val = np.mean(diffs)
+    std_val = np.std(diffs)
+    min_val: float = np.min(diffs)
+    max_val: float = np.max(diffs)
+    range_val = max_val - min_val
+
+    stats_text = (
+        f"Mean: {mean_val}\n"
+        f"Std Dev: {std_val:.6f}s\n"
+        f"Range:   {range_val:.6f}s\n"
+        f"Min:     {min_val:.6f}s\n"
+        f"Max:     {max_val:.6f}s"
+    )
+
+    # Place text box in the upper right corner
+    plt.text(
+        0.95,
+        0.95,
+        stats_text,
+        transform=plt.gca().transAxes,
+        fontsize=8,
+        verticalalignment="top",
+        horizontalalignment="right",
+        bbox=dict(boxstyle="round,pad=0.5", fc="wheat", alpha=0.5),
+    )
+
+
+def lsl_timestamp_diffs(csv_filepath: str, start_value: int, end_value: int):
+    """
+    This function reads 'lsl_timestamp' from a CSV, calculates the differences,
+    and plots the results, highlighting every peak.
+    """
+    if not os.path.exists(csv_filepath):
```

Comment:
I'd pull the data loading and checks out of this method. We can expect a numpy array of the lsl_timestamps in this method, and add another method `load_csv` or perhaps you can do that in the main method below (just less easy to test / mock in the later case).